File: src/test/java/com/raychenon/hackerrank/SherlockAndAnagramsTest.java
package com.raychenon.hackerrank;

import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 18/1/19
 * Under "Dictionaries and Hashmaps"
 * https://www.hackerrank.com/challenges/sherlock-and-anagrams
 */
public class SherlockAndAnagramsTest {

    @Test
    public void assert_abba() {
        assertEquals(SherlockAndAnagrams.sherlockAndAnagrams("abba"), 4);
    }

    @Test
    public void assert_abcd() {
        assertEquals(SherlockAndAnagrams.sherlockAndAnagrams("abcd"), 0);
    }
    
    @Test
    public void assert_ifailuhkqq() {
        assertEquals(SherlockAndAnagrams.sherlockAndAnagrams("ifailuhkqq"), 3);
    }

    @Test
    public void assert_kkkk() {
        assertEquals(SherlockAndAnagrams.sherlockAndAnagrams("kkkk"), 10);
    }
    
}

File: src/test/kotlin/com/raychenon/kotlin/hackerrank/SherlockAndAnagramsTest.kt
package com.raychenon.kotlin.hackerrank

import org.junit.Assert
import org.junit.Test

/**
 * User: raychenon
 * Date: 19/1/19
 */
class SherlockAndAnagramsTest {

    @Test
    fun assert_abba() {
        Assert.assertEquals(SherlockAndAnagramsKotlin.sherlockAndAnagrams("abba"), 4)
    }

    @Test
    fun assert_abcd() {
        Assert.assertEquals(SherlockAndAnagramsKotlin.sherlockAndAnagrams("abcd"), 0)
    }

    @Test
    fun assert_ifailuhkqq() {
        Assert.assertEquals(SherlockAndAnagramsKotlin.sherlockAndAnagrams("ifailuhkqq"), 3)
    }

    @Test
    fun assert_kkkk() {
        Assert.assertEquals(SherlockAndAnagramsKotlin.sherlockAndAnagrams("kkkk"), 10)
    }

}